src/components/document/Document.tsx

```diff
@@ -27,24 +28,66 @@ import { Toolbar, ToolbarLink } from './Toolbar/Toolbar'
 
 export const Document = observer(() => {
     const { classes } = useStyles()
+    const containerRef = useRef<HTMLDivElement>(null)
+    const [isEventAdded, setIsEventAdded] = useState(false)
 
     const {
         user,
         fullPage,
         printMode,
         tagsStore: { tags, tagsLoading, tagsLocked, handleSpecialTagClick },
-        documentStore: { data, pathname, loading, collection, digestUrl, docRawUrl, thumbnailSrcSet, tab, handleTabChange },
+        documentStore: {
+            data,
+            ocrData: ocrDataStore,
+            pathname,
+            loading,
+            collection,
+            digestUrl,
+            docRawUrl,
+            thumbnailSrcSet,
+            tab,
+            handleTabChange,
+            documentSearchStore: { toggleSearchInput, query, metaSearchStore, textSearchStore, pdfSearchStore },
+        },
         searchStore: {
             searchResultsStore: { previewNextDoc, previewPreviousDoc },
         },
     } = useSharedStore()
 
+    useEffect(() => {
+        if (!data?.content) return
+        const content = [data.content.text]
+        if(ocrDataStore?.length) content.push(...ocrDataStore.map(({text}) => text))
+        textSearchStore.setTextContent(content)
+    }, [data, ocrDataStore, textSearchStore])
```

Comment:
is it possible to get rid of this `useEffect` and use `reaction` on `data.content` and `ocrData` in `documentSearchStore`

Reply:
Yes, I agree with this